Services/Administration/OJS.Services.Administration.Data/Implementations/ProblemGroupsDataService.cs

```diff
@@ -24,6 +24,10 @@ public ProblemGroupsDataService(DbContext problemGroups)
             this.DbSet
                 .Where(pg => pg.ContestId == contestId);
 
+        public IQueryable<ProblemGroup> GetAllVisibleByContest(int contestId)
+            => this.GetAllByContest(contestId)
+                .Where(pg => !pg.IsDeleted);
```

Comment:
Why `VIsible` is added to the name? I don't see any logic for filtering `IsVisible` contests...